### ISSUE TITLE
Bug: [@typescript-eslint/no-deprecated] Super call of deprecated constructor is not reported

### ISSUE THREAD

**Issue by tobbexiv** [typescript-eslint/typescript-eslint#10390]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.3&fileType=.tsx&code=MYGwhgzhAECC0G8CwAoa0D0AqL0ACAJgKYAOATkcGAC5EHRYarrAD2AdhNWQK7DWsyACgCUiAL6pJKVKEgwAQtCIAPWuwIx4yNNDadufAcLE706bLggALVjxD0KJQdWgAjHq4KsiMdq1dGZnMIHhIiE2DpaVkOLmgASwMwdmAieABeaHYiAHc4UQBuWINE5NSiJSyc-IUioA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=&tokens=false

### Repro Code

```TypeScript
class A {
  /** @deprecated */
  constructor() {}
}

class B extends A {
  constructor() {
    /** should report but does not */
    super()
  }
}

const instanceA = new A();
const instanceB = new B();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error"
  }
};
```

### tsconfig

```jsonc
{}
```

### Expected Result

The super call in class B is being reported by `@typescript-eslint/no-deprecated`.

### Actual Result

The super call is not reported and therefore usage of the deprecated constructor is not found.

**Comment by JoshuaKGoldberg** [comment]
😄 it has been a steady joy for me to see all the random syntaxes people keep finding for this rule...
